### PR TITLE
Adding basic ForICount Kernel testing.

### PR DESCRIPTION
**Pull request by mdavis36**
# Summary

- This PR is a part of the test refactor
- It does the following:
  - Adds basic ForICount tests to the functional kernel test suite. 
  - Tests Kernel ForICount across common exec policies and index types.
  - Supports testing StronglyTyped types.

### TODO:
- [x] Remove or enable resource testing with ForICount. I temporarily left the Resource stuff commented in case we want to test with / without resources. I assume it's probably not really necessary.

### CODE REVIEW

**Comment by rhornung67** [comment]
@mdavis36 is this ready for review?

**Comment by mdavis36** [comment]
> @mdavis36 is this ready for review?

Yes it is

**Comment by rhornung67** [comment]
Need to merge in develop.

**Review by artv3 — APPROVED** [review]

**Review by artv3 — APPROVED** [review]

File: test/functional/kernel/basic-single-icount-loop/test-kernel-basic-single-icount-loop.cpp.in [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+//
+// Header for tests in ./tests directory
+//
+// Note: CMake adds ./tests as an include dir for these tests.
+//
+#include "test-kernel-basic-single-icount-loop-segments.hpp"
+
+// Sequential execution policy types
+using SequentialKernelExecPols = camp::list<
```
Comment (rhornung67) [inline test/functional/kernel/basic-single-icount-loop/test-kernel-basic-single-icount-loop.cpp.in:25]: This is fine for now.  At some point, we should take a pass through the *.cpp.in files and see if it makes sense to consolidate the policy definitions in header files and share them.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**